storage/csi-snapshot.md

```diff
@@ -0,0 +1,120 @@
+---
+title: CSI Snapshots
+authors:
+  - "@chuffman"
+reviewers:
+  - "@gnufied”
+  - “@jsafrane”
+approvers:
+  - "@..."
+creation-date: 2019-09-05
+last-updated: 2019-09-25
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# csi-snapshots
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs] 
+
+## Summary
+We want to include the upstream CSI Snapshot feature in OpenShift.
+
+## Motivation
+
+* This feature has been requested by both the OCS and CNV teams.
+* Snapshotting is an incredibly useful function when it comes to system changes in development areas, as it allows rapid rollback to preview dev versions
+* Container development benefits from filesystems that provide snapshot functionality
+
+### Goals
+
+* Backport and include the external-snapshotter sidecar.
+
+* Package and ship a downstream image of the this sidecar.
+
+### Non-Goals
+
+* Offer application-consistent snapshots by providing pre/post snapshot hooks to freeze/unfreeze applications and/or unmount/mount file system.
+
+* Provide higher-level management, such as backing up and restoring a pod and statefulSet, and creating a consistent group of snapshots.
+
+## Proposal
+
+This feature has already been implemented upstream, and is described in https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190709-csi-snapshot.md . It is currently documented upstream at https://kubernetes.io/docs/concepts/storage/volume-snapshots/ .
+
+We will be releasing the `external-snapshotter` sidecar image to enable CSI Snapshot support in OCP as Developer Preview only. This sidecar can be used by internal teams, such as CNV or OCS, for their drivers.
+
+### User Stories [optional]
+
+Similar to how users and administrators provision PersistentVolume and PersistentVolumeClaim resources, VolumeSnapshotContent and VolumeSnapshot API resources will be included to create volume snapshots for users and administrators. A VolumeSnapshotContent is a snapshot taken from a volume in the cluster that has been provisioned by an administrator. It is a resource in the cluster, similar to a PersistentVolume. A VolumeSnapshot is a request for snapshot of a volume by a user. It is similar to a PersistentVolumeClaim.
+
+In addition to the above, VolumeSnapshotClass resources can be used to describe the “class” of storage when provisioning a VolumeSnapshot.
+
+#### Story 1
+
+As an OCS developer, I want to release the `external-snapshotter` sidecar image so that CSI drivers can utilize volume snapshots.
```

Comment:
Perhaps "... so that ~~CSI drivers~~ **users** can utilize snapshots"? Nobody cares about what the driver does under the hood, as long as users see snapshots and are able to restore them in the end.

Reply:
Adjusted this to say `users` instead of `CSI drivers`.